SNUTT-2022/SNUTT/Models/TimePlace.swift

```diff
@@ -29,16 +29,37 @@ struct TimePlace: Identifiable {
     }
 
     var endTimeDouble: Double {
-        return TimeUtils.getTimeInDouble(from: endTime)
+        endTimeDouble(compactMode: false)
     }
 
-    var duration: Double {
-        return endTimeDouble - startTimeDouble
+    func duration(compactMode: Bool) -> Double {
+        return endTimeDouble(compactMode: compactMode) - startTimeDouble
     }
 
     var preciseTimeString: String {
         return "\(day.veryShortSymbol)(\(startTime)~\(endTime))"
     }
+
+    private func endTimeDouble(compactMode: Bool) -> Double {
+        if compactMode && !isCustom {
+            return TimeUtils.getTimeInDouble(from: endTime.roundUpForCompactMode())
+        } else {
+            return TimeUtils.getTimeInDouble(from: endTime)
+        }
+    }
+}
+
+private extension String {
+    func roundUpForCompactMode() -> String {
+        var time = TimeUtils.getTime(from: self)
+        if time.minute == 15 {
+            time.minute = 30
+        } else if time.minute == 50 {
+            time.hour += 1
+            time.minute = 0
+        }
+        return time.toString()
```

Comment:
여기 45분에 끝나는 수업, 20분에 끝나는 수업 등에 대한 코너케이스가 빠져있어서 아래처럼 수정 제안드립니다
```suggestion
    func roundUpForCompactMode() -> String {
        var time = TimeUtils.getTime(from: self)
        if time.minute < 30 && time.minute > 0 {
            time.minute = 30
        } else if time.minute > 30 {
            time.hour += 1
            time.minute = 0
        }
        return time.toString()
    }
```